packages/venia-concept/src/components/ProductImageCarousel/thumbnailList.js

```diff
@@ -5,17 +5,25 @@ import { List } from '@magento/peregrine';
 import classify from 'src/classify';
 import Thumbnail from './thumbnail';
 import defaultClasses from './thumbnailList.css';
+import { mediaGalleryEntry } from 'src/shared/propShapes';
 
 class ThumbnailList extends Component {
     static propTypes = {
         classes: PropTypes.shape({
             root: PropTypes.string
         }),
-        items: PropTypes.arrayOf(PropTypes.object).isRequired
+        items: PropTypes.arrayOf(mediaGalleryEntry).isRequired
     };
 
     render() {
-        return <List renderItem={Thumbnail} {...this.props} />;
+        // linear-time sort is possible when we have a numeric 'position' prop
```

Comment:
Few notes:

1. The comment in the top of `render` doesn't seem necessary - let's remove it.
2. We may want to reconsider doing this in `render`. Could be memoized, but we can always address this later
3. It does not seem ideal to make the client-side do this sorting. Are we sure this list isn't already sorted when returned from the API? If not, we should log a feature request for it to be ordered.

Reply:
I just checked on my local and that list already comes back from the API sorted on `position`

Reply:
I did this because of the presence of a `position` member in the first place. In my experience, something like that means that collection order is not significant. I'm gonna ask about this real quick.

Reply:
Makes sense to me. I don't see a problem with this. If items have a `position` prop, that ought to be the sort order regardless of how the API returns them.

Though I agree it should at least be memoized a bit (i.e., cache only the last result of filter + reduce).

Reply:
I implemented this as a `shouldComponentUpdate`! Good for me.

I also checked with the GQL team, and they confirmed that the returned order of the collections is not guaranteed to be position order, and `position` should win. They should change it if they can, and we should map GQL query results somewhere in our data layer to ensure that array order IS significant, but that's a future optimization.

Lastly, the linear-time sort comment seems like a nice reassurance to me; it's much faster than the `O(nlogn)` of `Array.prototype.sort()`.